cli/flox/src/commands/init/mod.rs

```diff
@@ -426,82 +464,252 @@ impl From<ProvidedPackage> for PackageToInstall {
+            .map(|pkg| {
+                // Type-inference fails without the fully-qualified method call
+                <PackageResolutionInfo as Into<ProvidedPackage>>::into(pkg)
+            });
+        let Some(pkg) = pkg else {
+            tracing::debug!("no compatible default package");
+            return Ok(None);
+        };
+        pkg
+    } else {
+        tracing::debug!("using pkgdb to find default compatible package");
+        let query = Query {
+            rel_path: Some(rel_path),
+            semver: version,
+            limit: Some(1),
+            deduplicate: false,
+            ..Default::default()
+        };
+        let params = SearchParams {
+            manifest: None,
+            global_manifest: PathOrJson::Path(global_manifest_path(flox)),
+            lockfile: PathOrJson::Path(global_manifest_lockfile_path(flox)),
+            query,
+        };
+
+        let (mut results, _) = do_search(&params)?;
+        if results.results.is_empty() {
+            tracing::debug!("no compatible default package version found");
+            return Ok(None);
+        }
+        let pkg = results.results.swap_remove(0);
+        pkg.try_into()?
     };
+    tracing::debug!(
+        version = pkg.version.as_ref().unwrap_or(&"null".to_string()),
+        "found default package version"
+    );
+    Ok(Some(pkg))
+}
 
-    let (mut results, _) = do_search(&params)?;
+/// Get a package as if installed with `flox install {package}`
+async fn get_default_package(flox: &Flox, package: &AttrPath) -> Result<ProvidedPackage> {
+    let pkg = if let Some(client) = flox.catalog_client.as_ref() {
+        tracing::debug!(
+            package = package.to_string(),
+            "using catalog client to find default package"
+        );
+        let resolved_groups = client
+            .resolve(vec![PackageGroup {
+                descriptors: vec![PackageDescriptor {
+                    attr_path: package.to_string(),
+                    install_id: package.to_string(),
+                    version: None,
+                    allow_pre_releases: None,
+                    derivation: None,
+                }],
+                name: package.to_string(),
+                system: flox.system.clone(),
+            }])
+            .await?;
+        let pkg: Option<ProvidedPackage> = resolved_groups
+            .first()
+            .and_then(|pkg_group| pkg_group.pages.first())
+            .and_then(|page| page.packages.as_ref())
+            .and_then(|pkgs| pkgs.first().cloned())
+            .map(|pkg| {
+                // Type-inference fails without the fully-qualified method call
+                <PackageResolutionInfo as Into<ProvidedPackage>>::into(pkg)
+            });
+        let Some(pkg) = pkg else {
+            tracing::debug!("no default package found");
+            return Err(anyhow!("Flox couldn't find any versions of {package}"))?;
+        };
+        pkg
```

Comment:
suggestion: because the catalog doesn't allow fuzzy installs, `get_default_package` and `get_default_package_if_compatible` are doing the same thing. So it would make sense to factor out this chunk of code

Reply:
I don't think I follow the fuzzy install part, could you explain?

Reply:
`get_default_package_if_compatible` was meant to check if `nixpkgs#something` met a version requirement. `get_default_package` was meant to mimic `flox install something`. I believe currently those are the same with `pkgdb`, but that wasn't always for sure (e.g. if `flox install something` could install `flox install something_123`, which is what I meant by fuzzy). With the catalog, those two are always the same, so we don't need to treat them differently